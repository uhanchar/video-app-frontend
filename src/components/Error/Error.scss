@import "src/constants";

.error {
  display: flex;
  justify-content: center;

  .content {
    color: $error-color;
    font-size: 1.125rem;
    font-weight: 500;
  }
}
