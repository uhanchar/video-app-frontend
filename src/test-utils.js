import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

const reducerInitialState = {};

function render(
  ui,
  {
    initialState = reducerInitialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {},
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';

export { render };
