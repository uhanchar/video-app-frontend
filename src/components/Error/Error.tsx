import React, { FC } from 'react';
import { Grid } from '@material-ui/core';

import 'components/Error/Error.scss';

interface IError {
  content: string;
}

const Error: FC<IError> = ({ content }) => (
  <Grid className="error" item xs={12}>
    <div className="content" data-testid="error">{ content }</div>
  </Grid>
);

export default Error;
