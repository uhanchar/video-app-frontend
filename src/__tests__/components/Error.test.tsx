import React from 'react';

import { render as renderWithRedux } from 'test-utils';

import Error from 'components/Error/Error';

const errorMessage = 'Test error';

describe('test for Error component', () => {
  it('should render Error component', () => {
    const { asFragment } = renderWithRedux(<Error content={errorMessage} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render error message', () => {
    const { getByTestId } = renderWithRedux(<Error content={errorMessage} />);

    expect(getByTestId('error')).toHaveTextContent(errorMessage);
  });
});
