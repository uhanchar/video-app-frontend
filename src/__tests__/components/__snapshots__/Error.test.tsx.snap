// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Error component should render Error component 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root error MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="content"
      data-testid="error"
    >
      Test error
    </div>
  </div>
</DocumentFragment>
`;
